rclpy/src/rclpy/_rclpy.c

```diff
@@ -851,6 +852,85 @@ rclpy_count_subscribers(PyObject * Py_UNUSED(self), PyObject * args)
   return _count_subscribers_publishers(args, "subscribers", rcl_count_subscribers);
 }
 
+typedef rcl_ret_t (* rcl_get_info_by_topic_func)(
+  const rcl_node_t * node,
+  rcutils_allocator_t * allocator,
+  const char * topic_name,
+  bool no_mangle,
+  rmw_topic_info_array_t * info_array);
+
+static PyObject *
+_get_info_by_topic(
+  PyObject * args, const char * type,
+  rcl_get_info_by_topic_func rcl_get_info_by_topic)
+{
+  PyObject * pynode;
+  const char * topic_name;
+  PyObject * pyno_mangle;
+
+  if (!PyArg_ParseTuple(args, "OsO", &pynode, &topic_name, &pyno_mangle)) {
+    return NULL;
+  }
+
+  rcl_node_t * node = (rcl_node_t *)PyCapsule_GetPointer(pynode, "rcl_node_t");
+  if (!node) {
+    return NULL;
+  }
+  bool no_mangle = PyObject_IsTrue(pyno_mangle);
+  rcutils_allocator_t allocator = rcutils_get_default_allocator();
+  rmw_topic_info_array_t info_array = rmw_get_zero_initialized_topic_info_array();
```

Comment:
Is this cleaned up properly when `rcl_get_info_by_topic` and `rmw_topic_info_array_fini` are not OK?

Reply:
Ah! Good catch. Thanks!